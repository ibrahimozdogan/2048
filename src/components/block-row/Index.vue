<template>
    <div class="block-row">
        <Block
            v-for="(block, blockIndex) in blocks"
            :key="`${rowIndex}-${blockIndex}`"
            :value="block"
        />
    </div>
</template>

<script>
    import Block from './Block';

    export default {
        props: {
            /**
             * @property {string} rowIndex
             */
            rowIndex: {
                type: Number,
                required: true,
            },
            /**
             * @property {Array<Number>} blocks
             */
            blocks: {
                type: Array,
                default: [],
            },
        },

        components: { Block }
    }
</script>

<style lang="scss" scoped>
    .block-row {
        display: flex;
        position: relative;
    }
</style>