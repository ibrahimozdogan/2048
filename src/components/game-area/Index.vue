<template>
    <div class="game-area">
        <Header></Header>
        <Board></Board>
    </div>
</template>

<script>
    import Board from './Board';
    import Header from './Header';

    export default {
        components: {
            Board,
            Header
        }
    };
</script>

<style lang="scss" scoped>
    $center: center;

    .game-area {
        display: flex;
        justify-content: $center;
        flex-direction: column;
        align-items: $center;
    }
</style>
