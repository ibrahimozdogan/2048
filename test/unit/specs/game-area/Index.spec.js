import { shallowMount } from '@vue/test-utils'
import Index from '@/components/game-area/Index'

describe('@components/game-area/Index.vue', () => {
    describe('render', () => {
        it('renders correctly', () => {
            let wrapper = shallowMount(Index, {});
            expect(wrapper.element).toMatchSnapshot();
        });
    });
});
