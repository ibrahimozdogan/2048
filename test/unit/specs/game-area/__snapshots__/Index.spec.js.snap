// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/game-area/Index.vue render renders correctly 1`] = `
<div
  class="game-area"
>
  <header-stub />
   
  <board-stub />
</div>
`;
